tests/runner/test_runner.py

```diff
@@ -175,6 +178,47 @@ async def test_serve_prints_help_message_on_startup(self, capsys):
         assert "tired-flow/test_runner" in captured.out
         assert "$ prefect deployment run [DEPLOYMENT_NAME]" in captured.out
 
+    is_python_38 = sys.version_info[:2] == (3, 8)
+    is_python_39 = sys.version_info[:2] == (3, 9)
+    is_python_310 = sys.version_info[:2] == (3, 10)
+    is_pydantic_v2 = PYDANTIC_VERSION.startswith("2.")
+
+    @pytest.mark.xfail(
+        is_python_310 and is_pydantic_v2,
+        reason="Will fail on Python 3.10 with Pydantic V2",
+    )
+    @pytest.mark.xfail(
+        is_python_39,
+        reason="Will fail on Python 3.9 regardless of Pydantic version",
+    )
+    async def test_serve_typed_container_inputs_flow(self, capsys):
+        if self.is_python_38:
+
+            @flow
+            def type_container_input_flow(arg1: List[str]) -> str:
+                print(arg1)
+                return ",".join(arg1)
+
+        else:
+
+            @flow
+            def type_container_input_flow(arg1: list[str]) -> str:
+                print(arg1)
+                return ",".join(arg1)
+
+        await serve(
```

Comment:
i dont think we need serve to repro here, do we?

Reply:
oof, yeah I think it does! I just assumed our tests for non-served flows were robust enough that we already tested for this.

Reply:
@zzstoatzz Added a test for just defining and running a flow under `test_flows.py`. But I'll let you know when the tests all pass because it seems that Python 3.10 isn't the only version with trouble. 3.9 seems to have some issues but can't seem to reproduce that locally.